packages/location-management/src/ducks/locationHierarchy/index.ts

```diff
@@ -0,0 +1,149 @@
+
+/** action creator when adding a tree to store
+ *
+ * @param apiResponse - the raw hierarchy as received from opensrp
+ * @param treeId - the treeId to use while saving to the store
+ */
+export function fetchTree(
+  apiResponse: RawOpenSRPHierarchy,
+  treeId: string | null = null
+): FetchedTreeAction {
+  const tree = generateJurisdictionTree(apiResponse);
+  return {
+    treeByRootId: {
+      [treeId ? treeId : tree.model.id]: tree,
+    },
+    type: TREE_FETCHED,
+  };
+}
+
+/** clear all the trees */
+export function deforest(): DeforestAction {
+  return {
+    treeByRootId: {},
+    type: DEFOREST,
+  };
+}
+
+// **************************** reducer ****************************
+
+/** The store's slice state
+ */
+export interface TreeState {
+  treeByRootId: Dictionary<TreeNode> | {};
+}
+
+/** Create an immutable tree state */
+export type ImmutableTreeState = TreeState & SeamlessImmutable.ImmutableObject<TreeState>;
+
+/** starting state */
+export const initialState: ImmutableTreeState = SeamlessImmutable({
+  treeByRootId: {},
+});
+
+// the reducer function
+/**
+ *
+ * @param {Dictionary} state - the store
+ * @param {AnyAction} action - the redux action
+ * @returns {object} - updated state
+ */
+export function hierarchyReducer(
+  state: ImmutableTreeState | Dictionary = initialState,
+  action: TreeActionTypes
+) {
+  switch (action.type) {
+    case TREE_FETCHED:
+      return {
+        ...state,
+        treeByRootId: { ...state.treeByRootId, ...action.treeByRootId },
+      };
+    case DEFOREST:
+      return {
+        ...state,
+        treeByRootId: action.treeByRootId,
+      };
+    default:
+      return state;
+  }
+}
+
+/** prop filters to customize selector queries */
+export interface Filters {
+  rootJurisdictionId?: string[] /** specify which tree to act upon, undefined allows all */;
+}
+
+/** retrieve the rootJurisdiction value
+ *
+ * @param _ - the store
+ * @param props -  the filterProps
+ */
+export const getRootJurisdictionId = (_: Partial<Store>, props: Filters) =>
+  props.rootJurisdictionId;
+
+/** gets all trees key'd by the rootNodes id
+ *
+ * @param state - the store
+ * @param _ -  the filterProps
+ */
+export const getTreesByRootId = (state: Partial<Store>, _: Filters): Dictionary<TreeNode> =>
+  (state as Dictionary)[hierarchyReducerName].treeByRootId;
+
+/** factory that returns a selector to retrieve the tree(s) using their rootNode's ids */
+export const getTreesByIds = () =>
+  createSelector(getTreesByRootId, getRootJurisdictionId, (treesById, rootIds): TreeNode[] => {
+    const allTreesArray = values(treesById);
+    if (rootIds === undefined) {
+      return allTreesArray;
+    }
+    const treesOfInterest: TreeNode[] = [];
+    rootIds.forEach((rootId) => {
+      const treeOfInterest = treesById[rootId];
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (treeOfInterest) {
```

Comment:
I think treeOfInterest.length > 0 would ensure we don't disable @typescript-eslint/no-unnecessary-condition

Reply:
@kahummer treeOfInterest is not guaranteed to be an object, it could be undefined, in which case it would result in an error

Reply:
would adding an or not work, like so const treesOfInterest: TreeNode[] | undefined = [];

Reply:
@kahummer I agree, that should have worked but, I am currently not sure why it does not. by the way `treeOfInterest` is not an array, I am expecting an object if rootId key exists, otherwise it should be undefined.
![image](https://user-images.githubusercontent.com/28119869/104732645-968fbc80-574e-11eb-93e9-b80e5fc3d7c2.png)

Reply:
casting the type works, please re-review